gfx.cpp

```diff
@@ -4015,19 +4015,10 @@ class GfxInternal
             return GFX_SET_ERROR(kGfxResult_InvalidParameter, "Cannot dispatch using an invalid sbt object");
         if(!width || !height || !depth)
             return kGfxResult_NoError;  // nothing to dispatch
-        uint32_t const max_num_groups = 65535; // AMD doesn't allow to dispatch more than 65535 groups at once
-        if(width > max_num_groups || height > max_num_groups || depth > max_num_groups)
+        uint32_t const max_size = 2UL << 30;
+        if(width * height * depth > max_size)
         {
-            GfxResult result;
-            GfxBuffer args_buffer = allocateConstantMemory(3 * sizeof(uint32_t));
-            uint32_t *args        = (uint32_t *)getBufferData(args_buffer);
-            GFX_ASSERT(args != nullptr);
-            args[0] = width;
-            args[1] = height;
-            args[2] = depth;
-            result  = encodeDispatchRaysIndirect(sbt, args_buffer); // use indirect dispatch to workaround group limit
-            destroyBuffer(args_buffer);
-            return result;
+            return GFX_SET_ERROR(kGfxResult_InvalidParameter, "Cannot dispatch when total rays exceed 2^30");
```

Comment:
exceed**s**?

Reply:
Had to double check it in Word 😜 But its spelling and grammar thinks that `exceed` is correct

Reply:
If the rays werent plural then exceed would be i.e. "Cannot dispatch when total ray **count** exceed**s** 2^30". English really does make no sense